crates/cairo-coverage/src/input/filter.rs

```diff
@@ -0,0 +1,90 @@
+use crate::cli::IncludedComponent;
+use crate::data_loader::LoadedData;
+use crate::input::sierra_to_cairo_map::StatementOrigin;
+use regex::Regex;
+use std::collections::HashSet;
+use std::iter::once;
+use std::sync::LazyLock;
+
+pub static VIRTUAL_FILE_REGEX: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"\[.*?]").unwrap());
+const SNFORGE_TEST_EXECUTABLE: &str = "snforge_internal_test_executable";
+
+#[derive(Eq, PartialEq, Hash)]
+enum ItemLabel {
+    TestFunction,
+    UserFunction,
+    NonUserFunction,
+    Macro,
+}
+
+impl From<IncludedComponent> for ItemLabel {
+    fn from(included_component: IncludedComponent) -> Self {
+        match included_component {
+            IncludedComponent::TestFunctions => ItemLabel::TestFunction,
+            IncludedComponent::Macros => ItemLabel::Macro,
+        }
+    }
+}
+
+pub struct Filter {
```

Comment:
The name `filter` is too general. Let's make it more specific e.g. `CodeCategoryFilter`